### DISCUSSION TITLE
Better mod & game version sorting/organizing

### DISCUSSION THREAD

**Discussion by RandomGgames** [modrinth/.github#46] [Feature requests]
So currently, when you upload a new version of a mod, most people will split up the mod version and supported version. There are some mod devs that upload an updated version of their mod and include multiple files that are game version specific. These files, through the API, are indistinguishable from each other other than the name. An example of this can be seen on [autopath v1.4.9](https://modrinth.com/mod/autopath/version/1.4.9).

What I would suggest is that when a new version is created, multiple files can be uploaded and each file requires a minecraft version or versions. This data would be accessed through the API as a list of strings representing game versions. The current game versions does not need to change, but if say a new version supports 1.18-1.19.2, each file separately would say for what. 1.18-1.18.2, then another 1.19-1.19.4 for example. Continuing to use autopath as an example, it would change from this:
![image](https://user-images.githubusercontent.com/17207260/226531978-193b5715-8283-474f-aa99-4c5947301802.png)
To this
![image](https://user-images.githubusercontent.com/17207260/226533541-a4424014-3f13-491b-81f1-c5bc44c0b9a2.png)

And for the API, something like 
```
"files": [
			{
				"hashes": {
					"sha1": "b125c3ce7130886ab6be74c0c52163f73ececbdb",
					"sha512": "ca2517707cae041c6ed4c02c97365e4ac6685ad2222089b8f0ffa4c8cb48f33a78cce05a71b40a5e5019c4c74c9aded3063c5cc5c73ae09ff4fa9d5ae3570264"
				},
				"url": "https://cdn.modrinth.com/data/rYdrYotO/versions/EfPAPFIr/autopath-mc1.19.3-1.4.9.jar",
				"filename": "autopath-mc1.19.3-1.4.9.jar",
				"primary": true,
				"size": 1066329,
				"file_type": null
			}, ...
```
Would become
```
"files": [
			{
				"hashes": {
					"sha1": "b125c3ce7130886ab6be74c0c52163f73ececbdb",
					"sha512": "ca2517707cae041c6ed4c02c97365e4ac6685ad2222089b8f0ffa4c8cb48f33a78cce05a71b40a5e5019c4c74c9aded3063c5cc5c73ae09ff4fa9d5ae3570264"
				},
				"url": "https://cdn.modrinth.com/data/rYdrYotO/versions/EfPAPFIr/autopath-mc1.19.3-1.4.9.jar",
				"filename": "autopath-mc1.19.3-1.4.9.jar",
				"game_versions": [
					"1.19.3"
				],
				"primary": true,
				"size": 1066329,
				"file_type": null
			}, ...
```

**Comment by triphora** [comment]
This is the incorrect way of uploading versions and I will be in contact with @lebonq to have them fix this.

**Reply by lebonq** [reply]
Fixed !
To be fair it was not really obvious to me. The button to add a file is _Upload a version_ which, for me, was referring to a new **mod** version and not just add a new file for a newer MC version. Maybe a better wording would avoid this issue in the future ? (like _Upload  new files_)

**Comment by RandomGgames** [comment]
Well in that case, some for https://modrinth.com/mod/enchanted-vertical-slabs/version/1.6

**Reply by triphora** [reply]
Please report any other projects you find with this issue.

**Reply by RandomGgames** [reply]
Would the best reason be for poor description, invalid metadata or other?

**Reply by RandomGgames** [reply]
And would you like me to report both of these?

**Reply by triphora** [reply]
Invalid metadata. And if you could I would appreciate it.

**Reply by RandomGgames** [reply]
Np! Happy to help out :D
